StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetDeferredValidator.swift

```diff
@@ -48,12 +49,41 @@ struct PaymentSheetDeferredValidator {
         }
         try validatePaymentMethodId(intentPaymentMethod: setupIntent.paymentMethod, paymentMethod: paymentMethod)
     }
-    
+
     static func validatePaymentMethodId(intentPaymentMethod: STPPaymentMethod?, paymentMethod: STPPaymentMethod) throws {
         guard let intentPaymentMethod = intentPaymentMethod else { return }
         guard intentPaymentMethod.stripeId == paymentMethod.stripeId else {
+            if intentPaymentMethod.type == paymentMethod.type {
+                switch paymentMethod.type.identifier {
```

Comment:
We should add a comment explaining why we're validating fingerprints if the id is not equal so that future readers know about the Connect edge case.